File: macos-menubar/TaskFlowBar/Models/Todo.swift
import Foundation

struct Todo: Codable, Identifiable {
    let id: String
    let title: String
    let description: String?
    let status: String  // backlog | todo | ready_for_code | in_progress | review | done
    let priority: String  // high | medium | low
    let dueDate: String?
    let projectId: String?
    let parentId: String?
    let project: String?
    let sortOrder: Int?
    let doneAt: String?
    let deletedAt: String?
    let createdAt: String
    let updatedAt: String

    enum CodingKeys: String, CodingKey {
        case id, title, description, status, priority, project
        case dueDate = "due_date"
        case projectId = "project_id"
        case parentId = "parent_id"
        case sortOrder = "sort_order"
        case doneAt = "done_at"
        case deletedAt = "deleted_at"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }

    var isBacklog: Bool { status == "backlog" }
    var isTodo: Bool { status == "todo" }
    var isInProgress: Bool { status == "in_progress" }
    var isReview: Bool { status == "review" }
    var isDone: Bool { status == "done" }
    var isHighPriority: Bool { priority == "high" }
    var isMediumPriority: Bool { priority == "medium" }
    var isLowPriority: Bool { priority == "low" }
}

struct TodoListResponse: Codable {
    let todos: [Todo]
    let meta: TodoMeta
}

struct TodoMeta: Codable {
    let total: Int
    let limit: Int
    let offset: Int
}

struct TodayTodosResponse: Codable {
    let todos: [Todo]
    let date: String?
    let timezone: String?
}

File: src/validators/todo.ts
import { z } from "zod";

export const createTodoSchema = z.object({
  title: z.string().min(1).max(200),
  description: z.string().max(2000).optional(),
  status: z.enum(["backlog", "todo", "in_progress", "review", "done"]).default("backlog"),
  priority: z.enum(["high", "medium", "low"]).default("medium"),
  due_date: z.string().regex(/^\d{4}-\d{2}-\d{2}$/).optional(),
  project: z.string().max(100).optional(),
  project_id: z.string().nullable().optional(),
  parent_id: z.string().nullable().optional(),
  sort_order: z.number().int().default(0),
});

export const updateTodoSchema = createTodoSchema.partial();

export const reorderTodosSchema = z.object({
  items: z.array(z.object({
    id: z.string().min(1),
    sort_order: z.number().int().min(0),
    parent_id: z.string().nullable().optional(),
  })).min(1).max(100),
});

export const listTodosQuery = z.object({
  status: z.enum(["backlog", "todo", "in_progress", "review", "done"]).optional(),
  priority: z.enum(["high", "medium", "low"]).optional(),
  project: z.string().optional(),
  project_id: z.string().optional(),
  sort: z.enum(["due_date", "priority", "created_at", "sort_order"]).default("sort_order"),
  order: z.enum(["asc", "desc"]).default("asc"),
  limit: z.coerce.number().int().min(1).max(1000).default(50),
  offset: z.coerce.number().int().min(0).default(0),
});
